viz/cmd/uninstall.go

```diff
@@ -38,5 +38,10 @@ func uninstallRunE(ctx context.Context) error {
 		return err
 	}
 
-	return pkgCmd.Uninstall(ctx, k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, ExtensionName))
+	err = pkgCmd.Uninstall(ctx, k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, ExtensionName))
+	if err == nil {
+		return nil
+	}
```

Comment:
I'm not quite sure this would work well, given that the `Uninstall()` command has more than one point of failure. An uninstall in this case may fail if the resources are not found, the clientset cannot be initialised properly or the resources cannot be rendered. 

In case of an error, by this logic we will re-attempt the uninstall but with the legacy label instead of the current one. I think we should either handle the error returned by `Uninstall` a bit more gracefully here; this would help us avoid any confusion down the line and adhere a bit more to Go's failure handling standards. 

Dealing with the error gracefully here can be a bit tedious (since we don't have different error types) so instead here is what I would do:

```suggestion
        extensionReq, err := labels.NewRequirement(k8s.LinkerdExtensionLabel, selection.In, []string{ExtensionName, LegacyExtensionName})
        if err != nil {
                return err
        }

        selector := labels.NewSelector().Add(*extensionReq)
        return pkgCmd.Uninstall(ctx, k8sAPI, selector.String())

```

We can make a bit more use of the selector provided by client-go (` "k8s.io/apimachinery/pkg/labels"`) and check that our key (`linkerd.io/extension` in this case) is in ['viz', 'linkerd-viz']. We can then turn the selector into a string and pass it to `Uninstall`.

An alternative way of doing this would be to build a label selector with the two values, turn it into a map and then finally into a string. Client-go again offers a few functions to make the conversion easier but this just seems a bit too complicated imo.

Let me know what you think.

Reply:
My change certainly didn't feel great, especially since it was an all errors case, and I didn't have a great sense of how to distinguish.

I will take anything that works. I'm not remotely picky.

Please pretend that I don't really speak golang or k8s, which is a sufficiently good approximation.



(I think one test may have failed which was a good indication that my quick hack wasn't great.)

In GitHub, the indentation of your changes looks off, not sure why.